### DISCUSSION TITLE
handling dates function

### DISCUSSION THREAD

**Discussion by harjeet-glean** [duckdb/duckdb#10734] [Q&A] [answered]
hi team i have a epoch timestamp which i want to convert to date format 'yyyy-MM-dd' and then subtract 90 days from it.
The following works 
`strftime('%Y-%m-%d', timestamp '1970-01-01' + 1630828200 * interval '1 second')
`
but the following doesn't and gives an error **CatalogException: Catalog Error: Scalar Function with name date does not exist!**

`duckdb.sql("SELECT DATE(strftime('%Y-%m-%d', timestamp '1970-01-01' + 1630828200 * interval '1 second')) - 90 ").show()`

Also isn't there an easy way, If i have trino/spark sql's making them to this format is hell lot of work and tricky , wondering are there easier functions like **from_unixtime** and **date_sub** like there are for other languages.

**Comment by cmdlineluser** [comment]
All the functions are [documented here](https://duckdb.org/docs/sql/functions/overview#pages-in-this-section) e.g.

- https://duckdb.org/docs/sql/functions/timestamp

The approach I've been using is:

```python
duckdb.sql("""
select epoch_ms(1630828200::bigint * 1000) - '90d'::interval
""").show()
```

```python
┌───────────────────────────────────────────────────────────────────────────┐
│ (epoch_ms((CAST(1630828200 AS BIGINT) * 1000)) - CAST('90d' AS INTERVAL)) │
│                                 timestamp                                 │
├───────────────────────────────────────────────────────────────────────────┤
│ 2021-06-07 07:50:00                                                       │
└───────────────────────────────────────────────────────────────────────────┘
```

**Reply by harjeet-glean** [reply]
well, you are still not getting the data in yyyy-MM-dd format. anyways I figured it out. this worked for me:
`cast(strftime('%Y-%m-%d', timestamp '1970-01-01' + 1630828200 * interval '1 second') as DATE) -90
`
but the point is this is not pretty, and very cumbersome to use. If I have complex sqls which I want to migrate to duckdb, this makes it very tricky and tough, ideally there should be standard functions like **date_format** that makes jobs easier for you.

**Reply by cmdlineluser** [reply, marked as answer]
```python
duckdb.sql("""
select (epoch_ms(1630828200::bigint * 1000) - '90d'::interval)::date
""").show()
```

```python
┌─────────────────────────────────────────────────────────────────────────────────────────┐
│ CAST((epoch_ms((CAST(1630828200 AS BIGINT) * 1000)) - CAST('90d' AS INTERVAL)) AS DATE) │
│                                          date                                           │
├─────────────────────────────────────────────────────────────────────────────────────────┤
│ 2021-06-07                                                                              │
└─────────────────────────────────────────────────────────────────────────────────────────┘
```
